dev/integration_tests/flutter_driver_screenshot_test/test_driver/driver_screenshot_tester.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io' show File;
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as path;
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:collection/collection.dart';
+
+const String _kPathParent = 'test_driver/goldens/';
+
+/// The utility class that helps test cases to tests screenshots with a [FlutterDriver].
+class DriverScreenShotTester {
+  /// Constructs a [DriverScreenShotTester].
+  ///
+  /// All the parameters are required and must not be null.
+  // ignore: always_require_non_null_named_parameters
+  DriverScreenShotTester(
+      {@required this.testName,
+      @required this.driver,
+      @required this.deviceModel})
+      : assert(testName != null),
+        assert(driver != null),
+        assert(deviceModel != null);
+
+  /// The name of the test.
+  ///
+  /// It needs to match the folder name which the goldens resides under `test_driver/goldens`.
+  final String testName;
+
+  /// The `FlutterDriver` used to take the screenshots.
+  final FlutterDriver driver;
+
+  /// The device model of the device that the test is running on.
+  final String deviceModel;
+
+  /// Compares `screenshot` to the corresponding golden image. Returns true if they match.
+  ///
+  /// The golden image should exists at `test_driver/goldens/<testName>/<deviceModel>.png`
+  /// prior to this call.
+  Future<bool> compareScreenshots(List<int> screenshot) async {
+    final File file = File(_getImageFilePath());
+    final List<int> matcher = file.readAsBytesSync();
```

Comment:
Might as well use the async version since you're already in an async method.

Reply:
done